src/metabase/task/sync_databases.clj

```diff
@@ -72,6 +74,30 @@
          end-time
          (< (.toMinutes (t/duration start-time end-time)) analyze-duration-threshold-for-refingerprinting))))
 
+(defn- sync-and-analyze-database*!
```

Comment:
I extracted the bulk of this out from `sync-and-analyze-database!` so we could test it.